### PR TITLE
Various doc updates

### PR DESCRIPTION
**Pull request by jherland**
- `README.md`: Add documentation about `venv` config file option
- `README.md`: Reword the section on contributing to our real tests
- `docs/CodeDesign.md`: Rewrite/update pass

### CODE REVIEW

File: docs/CodeDesign.md [hunk 1]
```diff
@@ -23,26 +26,36 @@ We want the code to be readable, testable and maintainable. That is why we use:
...
+for a good introduction to the code review culture we want to foster in this
+project.
 
 ## Tests
 
-We do not aim for 100% coverage but to document the usecases via tests. FawltyDeps has unit and integration tests.
+FawltyDeps has unit and integration tests.
 
-Tests have following naming convention:
+We do not specifically aim for 100% coverage but we do want to document the use
```
Comment (vreuter) [inline docs/CodeDesign.md:44]: Nit

```suggestion
We do not specifically aim for 100% coverage, but we do want to document the use
```

File: docs/CodeDesign.md [hunk 2]
```diff
@@ -23,26 +26,36 @@ We want the code to be readable, testable and maintainable. That is why we use:
...
+cases via tests.
 
+Many tests have following naming convention:
```
Comment (vreuter) [inline docs/CodeDesign.md:47]: Nit:

```suggestion
Many tests use following naming convention:
```

File: docs/CodeDesign.md [hunk 3]
```diff
@@ -23,26 +26,36 @@ We want the code to be readable, testable and maintainable. That is why we use:
 - `pytest` for testing
   to ensure this quality.
 
-The code must be type-annotated and functions should have docstrings.
+The code should be type-annotated and functions should have docstrings.
+
+FawltyDeps searches for imports and dependencies and to do it efficiently,
```
Comment (vreuter) [inline docs/CodeDesign.md:31]: Nit:

```suggestion
FawltyDeps searches for imports and dependencies, and to do it efficiently,
```

File: docs/CodeDesign.md [hunk 4]
```diff
@@ -1,20 +1,23 @@
...
+project, and immediately get some useful and actionable information.
 
-- Automatically discover all Python code files and extract all the packages imported in the code.
-- Automatically discover and extract from files declaring dependencies (supporting requirements.txt, setup.py and pyproject.toml).
-- Use identity mapping between extracted imports and declared dependencies,
-- Compare the two sets (of used vs declared dependencies) and report missing and unused dependencies.
-- Support Python version 3.7 - 3.11
+## Boundaries
+
+- Support all current Python versions, that means all Python versions that have
```
Comment (vreuter) [inline docs/CodeDesign.md:11]: Nit:

```suggestion
- Support all current Python versions: that means all Python versions that have
```

File: docs/CodeDesign.md [hunk 5]
```diff
@@ -1,20 +1,23 @@
 # Requirements
 
-## Prototype
+## Initial goals
 
-The goal of the first prototype is to reach a working and usable tool with minimal functionality.
-Namely, a user would be able to run `fawltydeps` as a command line tool in their Python project.
-FawltyDeps will:
+We want a working and usable tool that slowly grows functionality as needed.
```
Comment (vreuter) [inline docs/CodeDesign.md:5]: Nit:

```suggestion
We want a working and usable tool that slowly adds functionality as needed.
```

File: docs/CodeDesign.md [hunk 6]
```diff
@@ -1,20 +1,23 @@
...
+A user would be able to run `fawltydeps` as a command line tool in their Python
```
Comment (vreuter) [inline docs/CodeDesign.md:6]: I think it makes sense and reads more naturally if we match verb tense across these items. I don't have a strong preference about _which_ tense, just agreement among them

File: README.md [hunk 7]
```diff
@@ -231,6 +231,9 @@ Here is a complete list of configuration directives we support:
   Defaults to the current directory, i.e. like `code = ["."]`.
 - `deps`: Files or directories containing the declared dependencies.
   Defaults to the current directory, i.e. like `deps = ["."]`.
+- `venv`: The path to the virtualenv to use for resolving project dependencies
```
Comment (vreuter) [inline README.md:234]: I guess this will need updated based on our morning meeting, maybe `pyenv` ?

File: README.md [hunk 8]
```diff
@@ -231,6 +231,9 @@ Here is a complete list of configuration directives we support:
...
+  into provided import names. Defaults to the Python environment where
+  FawltyDeps is installed).
```
Comment (vreuter) [inline README.md:236]: It looks like this closing parenthesis isn't matched to an opening one?

**Review by vreuter — CHANGES_REQUESTED** [review]
overall looks good, just a lot of Nits by me and a couple of questions

File: docs/CodeDesign.md [hunk 9]
```diff
@@ -1,20 +1,23 @@
...
-- Automatically discover and extract from files declaring dependencies (supporting requirements.txt, setup.py and pyproject.toml).
-- Use identity mapping between extracted imports and declared dependencies,
-- Compare the two sets (of used vs declared dependencies) and report missing and unused dependencies.
-- Support Python version 3.7 - 3.11
+## Boundaries
+
+- Support all current Python versions, that means all Python versions that have
+  a stable release, and are not yet End Of Life. Currently this is: v3.7 - v3.11.
+- For now we support the CPython interpreter only
+- OS-wise, for now we have concentrated on Linux. We should still run fine on
```
Comment (Nour-Mws) [inline docs/CodeDesign.md:14]: Repetition of "for now" that is a bit jarring

File: docs/CodeDesign.md [hunk 10]
```diff
@@ -1,20 +1,23 @@
...
+  other Unix-like systems, most notably Mac. Windows is still an open question.
```
Comment (Nour-Mws) [inline docs/CodeDesign.md:15]: Same for still.

File: docs/CodeDesign.md (hunk 2 above)
Comment (Nour-Mws) [inline docs/CodeDesign.md:47]:
```suggestion
Many tests have the following naming convention:
```

File: README.md [hunk 11]
```diff
@@ -355,17 +358,23 @@ while verifying their output. These projects, along with the expected FawltyDeps
 outputs, are defined in TOML files under
 [`tests/real_projects`](./tests/real_projects).
 
-## Contributing
+### Contributing more projects to our test suite
```
Comment (mknorps) [inline README.md:361]:
```suggestion
### Contributing more projects to the test suite
```
more inclusive

File: docs/CodeDesign.md [hunk 12]
```diff
@@ -23,26 +26,36 @@ We want the code to be readable, testable and maintainable. That is why we use:
...
 
+Many tests have following naming convention:
 ```
 test_{tested_function}__{short_description}__{expected_result}
 ```
 
 ## Class hierarchy
 
-Our classes can be viewed as four "layers":
+Our main classes can be viewed as four "layers":
```
[the remark this answers is not in the document]
Comment (jherland) [reply docs/CodeDesign.md:54]:
```suggestion
Our main classes can be roughly "layered" as follows:
```

File: docs/CodeDesign.md [hunk 13]
```diff
@@ -23,26 +26,36 @@ We want the code to be readable, testable and maintainable. That is why we use:
...
-The code must be type-annotated and functions should have docstrings.
+The code should be type-annotated and functions should have docstrings.
+
+FawltyDeps searches for imports and dependencies and to do it efficiently,
+generators are used in many places.
+
+## Code review
 
-FawltyDeps searches for imports and dependencies and to do it efficiently, generators are used.
+Read https://m.mediawiki.org/wiki/Guidelines_for_a_healthy_code_review_culture
```
[the remark this answers is not in the document]
Comment (jherland) [reply docs/CodeDesign.md:36]: Thanks!

```suggestion
Read https://www.mediawiki.org/wiki/Guidelines_for_a_healthy_code_review_culture
```

File: docs/CodeDesign.md (hunk 6 above)
Comment (jherland) [reply docs/CodeDesign.md:6]:
```suggestion
A user runs `fawltydeps` as a command line tool in their Python
```

File: docs/CodeDesign.md (hunk 9 above)
Comment (jherland) [reply docs/CodeDesign.md:14]:
```suggestion
- For now we support the CPython interpreter only
- OS-wise, we have concentrated on Linux. We should still run fine on
```

File: docs/CodeDesign.md (hunk 10 above)
Comment (jherland) [reply docs/CodeDesign.md:15]:
```suggestion
- OS-wise, for now we have concentrated on Linux. We should still run fine on
  other Unix-like systems, most notably Mac. Windows remains open question.
```

File: docs/CodeDesign.md [hunk 14]
```diff
@@ -1,20 +1,23 @@
...
-- Use identity mapping between extracted imports and declared dependencies,
-- Compare the two sets (of used vs declared dependencies) and report missing and unused dependencies.
-- Support Python version 3.7 - 3.11
+## Boundaries
+
+- Support all current Python versions: that means all Python versions that have
+  a stable release, and are not yet End Of Life. Currently this is: v3.7 - v3.11.
+- For now we support the CPython interpreter only
+- OS-wise, we have concentrated on Linux. We should still run fine on
+  other Unix-like systems, most notably Mac. Windows is still an open question.
```
Comment (jherland) [inline docs/CodeDesign.md:15]:
```suggestion
  other Unix-like systems, most notably Mac. Windows remains an open question.
```

File: docs/CodeDesign.md [hunk 15]
```diff
@@ -1,20 +1,23 @@
...
 
-## Prototype
+## Initial goals
 
-The goal of the first prototype is to reach a working and usable tool with minimal functionality.
-Namely, a user would be able to run `fawltydeps` as a command line tool in their Python project.
-FawltyDeps will:
+We want a working and usable tool that slowly adds functionality as needed.
+A user runs `fawltydeps` as a command line tool in their Python
+project, and immediately get some useful and actionable information.
```
Comment (vreuter) [inline docs/CodeDesign.md:7]:
```suggestion
project and immediately gets some useful and actionable information.
```

**Review by vreuter — APPROVED** [review]
just a small grammar change, then good to :rocket:

**Pull request merged**